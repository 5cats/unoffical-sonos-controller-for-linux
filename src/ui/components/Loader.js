import { h } from 'preact';
import { connect } from 'preact-redux';

import { getHasCurrent } from '../selectors/LoaderSelectors';

const mapStateToProps = state => {
    return {
        hasCurrent: getHasCurrent(state)
    };
};

const mapDispatchToProps = () => {
    return {};
};

export function Loader(props) {
    if (props.hasCurrent) {
        return null;
    }

    return (
        <div id="loader">
            <p>Searching for your Sonos System ...</p>
        </div>
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(Loader);
