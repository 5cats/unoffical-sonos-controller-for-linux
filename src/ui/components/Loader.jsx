import React from 'react';
import { connect } from 'react-redux';

import { getHasCurrent } from '../selectors/LoaderSelectors';

const mapStateToProps = (state) => {
    return {
        hasCurrent: getHasCurrent(state),
    };
};

const mapDispatchToProps = () => {
    return {};
};

export function Loader(props) {
    if (props.hasCurrent) {
        return null;
    }

    return (
        <div id="loader">
            <p>Searching for your Sonos System ...</p>
        </div>
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(Loader);
