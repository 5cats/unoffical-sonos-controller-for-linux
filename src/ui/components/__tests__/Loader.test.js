import { h } from 'preact';
import { Loader } from '../Loader';
import { deep } from 'preact-render-spy';

describe('Loader', () => {
    it('renders when props.hasCurrent is false', () => {
        const context = deep(<Loader />);
        expect(context.find('div').length).toBe(1);
        expect(context.output()).toMatchSnapshot();
    });

    it('renders nothing is true', () => {
        const context = deep(<Loader hasCurrent={true} />);
        expect(context.find('div').length).toBe(0);
        expect(context.output()).toMatchSnapshot();
    });
});
