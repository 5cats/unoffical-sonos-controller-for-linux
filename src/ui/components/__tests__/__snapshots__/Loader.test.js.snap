// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders nothing is true 1`] = `null`;

exports[`Loader renders when props.hasCurrent is false 1`] = `
<div id="loader">
  <p>Searching for your Sonos System ...</p>
</div>
`;
